### PR TITLE
Fix no route_short generation

### PR DESCRIPTION
**Pull request by eifinger**
The previous PR had a bug in it which lead to no route generation at all.

If this PR is merged, can you release a new version?

### CODE REVIEW

File: herepy/routing_api.py [hunk 1]
```diff
@@ -60,16 +60,16 @@ def _route(self, waypoint_a, waypoint_b, modes=None):
         route = response.response["route"]
         maneuver = route[0]["leg"][0]["maneuver"]
 
-        if any(mode in modes for mode in [RouteMode.car, RouteMode.truck]):
+        if any(str(mode) in modes for mode in [RouteMode.car, RouteMode.truck]):
```
Comment (abdullahselek) [inline herepy/routing_api.py]: For each enum we have a `__str__` method which returns the str value of enum type, I guess it's better to use that.

Comment (abdullahselek) [reply herepy/routing_api.py]: If you agree please update other `str(mode)` usages as well.

Comment (eifinger) [reply herepy/routing_api.py]: This is already used according to the str() [documentation](https://docs.python.org/3/library/stdtypes.html#str):
```text
[...]If neither encoding nor errors is given, str(object) returns object.__str__()[...]
````

Comment (abdullahselek) [reply herepy/routing_api.py]: Please check `RouteMode` for some of the types string value is different like `traffic_disabled = 'traffic:disabled'` it's better to use `__str(mode)__` to get the correct value.

**Comment by eifinger** [comment]
Turns out the problem was that I was using the direct strings to construct the modes instead of using the RouteMode enum.

**Pull request closed without merging**